pyramid/config/views.py

```diff
@@ -933,7 +933,7 @@ def register(permission=permission, renderer=renderer):
             if not exception_only and isexc:
                 derived_view = runtime_exc_view(derived_view, derived_exc_view)
 
-            derived_view.__discriminator__ = lambda *arg: discriminator
+            derived_view.__discriminator__ = lambda *arg: view_intr.discriminator
```

Comment:
Would this be solved if we fixed the `Deferred` to release the closure when it is resolved? Imagine `self.func = None` in https://github.com/Pylons/pyramid/blob/7fb889349d40e361397ed31f3535689182aa5cf0/pyramid/registry.py#L278